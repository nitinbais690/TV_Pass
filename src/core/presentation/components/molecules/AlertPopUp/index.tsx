import React from 'react';
import { Text, Modal, View } from 'react-native';
import { popupStyle } from './styles';
import FlexButtons, { FlexButtonsProps } from '../../atoms/FlexButtons';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import LinearGradient from 'react-native-linear-gradient';
import AlertIcon from 'assets/images/alert_icon.svg';

const AlertPopUp = (props: AlertPopUpProps): JSX.Element => {
    const styles = popupStyle(useAppColors());

    return (
        <Modal
            animationType="fade"
            transparent={true}
            statusBarTranslucent={true}
            visible={props.isVisible}
            onRequestClose={() => {
                props.onModelClosed();
            }}>
            <View style={styles.container}>
                <LinearGradient
                    colors={['#3B4046', '#2D3037']}
                    start={{ x: 0, y: 0 }}
                    end={{ x: 0, y: 1 }}
                    style={styles.gradientContainer}>
                    <View style={styles.textContainer}>
                        <LinearGradient
                            colors={['rgba(78, 67, 63, 0.5)', 'rgba(255, 110, 69, 0.5)']}
                            start={{ x: 0, y: 0 }}
                            end={{ x: 0.5, y: 0.5 }}
                            style={styles.iconContainer}>
                            <AlertIcon />
                        </LinearGradient>
                        <Text style={styles.textStyle}>{props.alertMessage}</Text>
                    </View>

                    <FlexButtons {...props} containerStyle={[props.containerStyle, styles.buttons]} />
                </LinearGradient>
            </View>
        </Modal>
    );
};

interface AlertPopUpProps extends FlexButtonsProps {
    isVisible: boolean;
    onModelClosed: any;
    showAlertIcon?: boolean;
    alertMessage: string;
}

export default AlertPopUp;
