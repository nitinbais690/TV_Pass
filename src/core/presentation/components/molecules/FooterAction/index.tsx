import React from 'react';
import { View } from 'react-native';
import PrimaryButton from '../../atoms/PrimaryButton';
import SecondaryButton from '../../atoms/SecondaryButton';
import { FooterActionStyle } from './style';

const FooterAction = (props: FooterActionProps): JSX.Element => {
    const styles = FooterActionStyle();

    return (
        <View style={styles.container}>
            {props.secondryButton && (
                <SecondaryButton
                    containerStyle={[styles.secondaryButton]}
                    title={props.secondryButton.label}
                    onPress={props.secondryButton.onPress}
                    secondaryTestID={props.secondryButton.testId}
                    secondaryAccessibilityLabel={props.secondryButton.accessibilityLabel}
                />
            )}
            <PrimaryButton
                containerStyle={[styles.primaryButton]}
                title={props.primartButton.label}
                onPress={props.primartButton.onPress}
                primaryTestID={props.primartButton.testId}
                primaryAccessibilityLabel={props.primartButton.accessibilityLabel}
            />
        </View>
    );
};

export default FooterAction;

export interface FooterActionProps {
    primartButton: FooterButtonProps;
    secondryButton?: FooterButtonProps;
}
export interface FooterButtonProps {
    label: string;
    testId: string;
    accessibilityLabel: string;
    onPress: () => void;
}
