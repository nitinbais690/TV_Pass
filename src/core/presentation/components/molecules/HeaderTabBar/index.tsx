import React, { useRef, useEffect } from 'react';
import { View, TouchableOpacity, Animated, FlatList, Text, Platform } from 'react-native';
import { BorderlessButton } from 'react-native-gesture-handler';
import { useSafeArea } from 'react-native-safe-area-context';
import DeviceInfo from 'react-native-device-info';
import { SceneRendererProps, NavigationState } from 'react-native-tab-view';
import { useNavigation } from '@react-navigation/native';
import { useAppPreferencesState } from 'utils/AppPreferencesContext';
import { HeaderContextProvider } from 'contexts/HeaderContextProvider';
import LinearGradient from 'react-native-linear-gradient';
// import SearchIcon from 'assets/images/search.svg';
import NotificationIcon from 'assets/images/notification.svg';
import CatalogueIcon from 'assets/images/catalogue-icon.svg';
import PagerView from 'react-native-pager-view';

import { scale, selectDeviceType } from 'qp-common-ui';
import { NAVIGATION_TYPE } from 'screens/Navigation/NavigationConstants';
import { useAlert } from 'contexts/AlertContext';
import { useLocalization } from 'contexts/LocalizationContext';
import HeaderTitle from 'core/presentation/components/atoms/HeaderTitle';
import { headerStyles, tabStyles, LANGUAGE_ICON_SIZE, NOTIFICATION_ICON_SIZE } from './styles';

const HeaderTabBar = <T extends Route>(props: HeaderTabBarProps<T>): JSX.Element => {
    const navigation = useNavigation();
    const insets = useSafeArea();
    const prefs = useAppPreferencesState();
    let { appColors } = prefs.appTheme!(prefs);
    const { strings } = useLocalization();
    const { Alert } = useAlert();
    const [index, setIndex] = React.useState(props.initialTab || 0);

    const ref = useRef<PagerView>(null);

    const autoWidth = props.autoWidth || selectDeviceType({ Handset: false }, true);
    const topInset = insets.top ? insets.top : 22; //include Status bar height for non-notch devices.

    const headerStylesSet = headerStyles(topInset, appColors);
    const tabStylesSet = tabStyles(props, autoWidth);

    const showComingSoonAlert = React.useCallback((): void => {
        Alert.alert(
            'Coming Soon',
            undefined,
            [
                {
                    text: strings['global.okay'],
                },
            ],
            {
                cancelable: false,
            },
        );
    }, [Alert, strings]);

    const setHeaderInset = () => {};
    const onPress = (index: number) => {
        if (ref.current) {
            ref.current.setPage(index);
        }
        setIndex(index);
    };

    // Fix to load initial page on PagerView
    useEffect(() => {
        if (props.routes && props.routes.length > 0 && ref.current && Platform.OS === 'ios') {
            ref.current.setPageWithoutAnimation(0);
        }
    }, [props.routes]);

    const tabItem = (item: any) => {
        return (
            <TouchableOpacity
                key={item.key}
                onPress={() => {
                    onPress(item.index);
                }}>
                <View>
                    {item.index === index ? (
                        <View>
                            <LinearGradient
                                colors={['#3B4046', '#2D3037']}
                                locations={[0.09, 1]}
                                style={[
                                    tabStylesSet.shadow,
                                    tabStylesSet.tabBarLabelStyleSelected,
                                    tabStylesSet.tabBarLabel,
                                ]}>
                                <Text style={[tabStylesSet.tabBarLabelStyle, tabStylesSet.tabBarLabelStyleSelected]}>
                                    {item.item.data.name}
                                </Text>
                            </LinearGradient>
                        </View>
                    ) : (
                        <Animated.View style={[tabStylesSet.tabBarLabel]}>
                            <Text style={[tabStylesSet.tabBarLabelStyle]}>{item.item.data.name}</Text>
                        </Animated.View>
                    )}
                </View>
            </TouchableOpacity>
        );
    };

    const getPager = () => {
        let output = [];
        for (let i = 0; i < props.routes.length; i++) {
            const element = props.routes[i];
            output.push(props.renderScenePager({ route: element, index: i }));
        }
        return (
            <PagerView ref={ref} scrollEnabled={false} style={[headerStylesSet.pagerContainerStyle]} initialPage={0}>
                {output}
            </PagerView>
        );
    };

    return (
        <HeaderContextProvider offset={20} onScrollEvent={setHeaderInset}>
            <Animated.View style={[headerStylesSet.headerContainerStyle]}>
                <Animated.View style={headerStylesSet.mobileContainer}>
                    <View style={headerStylesSet.headerLeftContainerStyle}>
                        <BorderlessButton
                            onPress={() => {
                                showComingSoonAlert();
                            }}>
                            <NotificationIcon width={NOTIFICATION_ICON_SIZE} height={NOTIFICATION_ICON_SIZE} />
                        </BorderlessButton>
                    </View>
                    <HeaderTitle />
                    <View style={headerStylesSet.headerRightContainerStyle}>
                        <BorderlessButton style={headerStylesSet.chromeCastIconStyle}>
                            {/* Chrome Cast Icon Will Come Here */}
                        </BorderlessButton>
                        <BorderlessButton
                            onPress={() => {
                                navigation.navigate(NAVIGATION_TYPE.SWITCH_LANGUAGE);
                            }}>
                            <CatalogueIcon width={LANGUAGE_ICON_SIZE} height={LANGUAGE_ICON_SIZE} />
                        </BorderlessButton>
                    </View>
                </Animated.View>
                <View style={tabStylesSet.tabBarContainerPager}>
                    <FlatList
                        horizontal={true}
                        pagingEnabled={true}
                        showsHorizontalScrollIndicator={false}
                        legacyImplementation={false}
                        data={props.routes}
                        renderItem={item => tabItem(item)}
                        keyExtractor={(item: T) => String(item.title)}
                    />
                </View>
                {getPager()}
            </Animated.View>
        </HeaderContextProvider>
    );
};

export default React.memo(HeaderTabBar);

export const useHeaderTabBarHeight = () => {
    let type = DeviceInfo.getDeviceType();
    const isHandset = type === 'Handset';
    const insets = useSafeArea();
    const topInset = insets.top ? insets.top : 22; //include Status bar height for non-notch devices.
    return isHandset ? scale(38) + topInset + scale(54) : scale(80);
};

export declare type Route = {
    key: string;
    icon?: string;
    title?: string;
    accessible?: boolean;
    accessibilityLabel?: string;
    testID?: string;
};

export interface HeaderTabBarProps<T extends Route> {
    routes: T[];
    renderScenePager: (props: { route: T; index: number }) => React.ReactNode;
    swipeEnabled?: boolean;
    autoWidth?: boolean;
    centerTabs?: boolean;
    scrollEnabled?: boolean;
    lazy?: boolean;
    lazyPreloadDistance?: number;
    removeClippedSubviews?: boolean;
    initialTab?: number;
}

export interface RNTabBarProps<T extends Route> {
    tabBarProps: SceneRendererProps & {
        navigationState: NavigationState<T>;
    };
}
