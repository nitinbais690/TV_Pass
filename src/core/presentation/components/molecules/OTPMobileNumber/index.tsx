import React from 'react';
import { View, Text } from 'react-native';

import { otpMobileNumberStyles } from './style';
import { useLocalization } from 'contexts/LocalizationContext';
import EditRow from 'core/presentation/components/atoms/EditRow';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import SubSectionLabel from 'core/presentation/components/atoms/SubSectionLabel';

export default function OTPMobileNumber(props: OTPMobileNumberProps) {
    const { strings } = useLocalization();
    const styles = otpMobileNumberStyles(useAppColors());
    return (
        <View style={[styles.container, props.style]}>
            <SubSectionLabel title={props.subSectionTitle} />
            <View style={styles.mobileNumberContainer}>
                <Text style={styles.mobileNumberText}>{props.mobileNumber}</Text>
                <EditRow showText={true} onPress={props.onChangePressed} editText={strings['verifyotp.change']} />
            </View>
        </View>
    );
}
export interface OTPMobileNumberProps {
    style?: {};
    subSectionTitle: string;
    mobileNumber: string;
    onChangePressed: () => void;
}
