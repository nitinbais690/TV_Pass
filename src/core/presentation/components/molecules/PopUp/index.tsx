import React, { useState } from 'react';
import { View, ScrollView, TouchableOpacity, Modal } from 'react-native';
import DetailsClose from 'assets/images/details_close.svg';
import DetailsBackground from 'features/details/presentation/components/atoms/DetailsBackground';
import { popupStyle } from './styles';
import { AUTOMATION_TEST_ID } from 'core/presentation/automation-ids';
import { MenuProvider } from 'react-native-popup-menu';

const PopUp = ({ onModelClosed, children }: PopUpProps): JSX.Element => {
    const styles = popupStyle();
    const [showPopup, setPopupVisibility] = useState(true);

    return (
        <Modal
            animationType="slide"
            transparent={true}
            statusBarTranslucent={true}
            visible={showPopup}
            onRequestClose={() => {
                onModelClosed();
                setPopupVisibility(false);
            }}>
            <MenuProvider skipInstanceCheck style={styles.menuProvider}>
                <View style={styles.container}>
                    <ScrollView style={styles.scrollViewContainer}>
                        <TouchableOpacity
                            style={styles.closeButton}
                            onPress={() => {
                                onModelClosed();
                                setPopupVisibility(false);
                            }}
                            testID={AUTOMATION_TEST_ID.CLOSE_ICON}
                            accessibilityLabel={AUTOMATION_TEST_ID.CLOSE_ICON}>
                            <DetailsClose width={'100%'} height={'100%'} />
                        </TouchableOpacity>

                        <DetailsBackground style={styles.gradientContainer}>{children}</DetailsBackground>
                    </ScrollView>
                </View>
            </MenuProvider>
        </Modal>
    );
};

interface PopUpProps {
    onModelClosed: any;
    children: React.ReactNode;
}

export default PopUp;
