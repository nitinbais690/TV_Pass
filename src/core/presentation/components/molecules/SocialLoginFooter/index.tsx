import React from 'react';
import { View, Text, Image } from 'react-native';
import { useLocalization } from 'contexts/LocalizationContext';
import SocialLoginButton from 'core/presentation/components/atoms/SocialLoginButton';
import EmailIcon from 'assets/images/social-login-icons/email.svg';
import GoogleIcon from 'assets/images/social-login-icons/google.svg';
import { socialLoginFooter } from './style';
import { useAppPreferencesState } from 'utils/AppPreferencesContext';
import { useNavigation } from '@react-navigation/native';
import { NAVIGATION_TYPE } from 'screens/Navigation/NavigationConstants';
import { EMAIL_AUTH_CONSTANTS, AUTH_TYPES } from 'features/authentication/utils/auth-constants';
import useGoogleSignin from 'core/presentation/hooks/useGoogleSignin';
import useFacebookSignin from 'core/presentation/hooks/useFacebookSignin';
import { AUTOMATION_TEST_ID } from 'core/presentation/automation-ids';

export default function SocialLoginFooter(props: SocialLoginFooterProps) {
    const [googleUser, initGoogleSignin] = useGoogleSignin();
    const [facebookUser, initFacebookSignin] = useFacebookSignin();
    const { strings }: any = useLocalization();
    const prefs = useAppPreferencesState();
    const navigation = useNavigation();
    const { appTheme } = prefs;
    let { appColors } = appTheme && appTheme(prefs);
    let socialLoginFooterStyle = socialLoginFooter(appColors);

    const providerGoogle = 'google';
    const providerFacebook = 'facebook';

    const onGoogleSignin = () => {
        props.onSignProgress();
        initGoogleSignin();
    };

    React.useEffect(() => {
        if (!googleUser.isSignedIn && googleUser.error) {
            props.onSigninError(googleUser.error, providerGoogle);
        }

        if (googleUser.isSignedIn && googleUser.id) {
            props.onSigninSuccess(googleUser.id, providerGoogle);
        }

        //eslint-disable-next-line react-hooks/exhaustive-deps
    }, [googleUser]);

    const onFacebookSignin = () => {
        props.onSignProgress();
        initFacebookSignin();
    };

    React.useEffect(() => {
        if (!facebookUser.isSignedIn && facebookUser.error) {
            props.onSigninError(facebookUser.error, providerFacebook);
        }

        if (facebookUser.isSignedIn && facebookUser.id) {
            props.onSigninSuccess(facebookUser.id, providerFacebook);
        }

        //eslint-disable-next-line react-hooks/exhaustive-deps
    }, [facebookUser]);

    return (
        <View style={socialLoginFooterStyle.container}>
            <Text style={socialLoginFooterStyle.footerText}>{strings['signin.footer.text']}</Text>
            <View style={socialLoginFooterStyle.socialLoginButtons}>
                <View
                    style={socialLoginFooterStyle.socialButtonStyle}
                    testID={AUTOMATION_TEST_ID.SOCIAL_BUTTON_EMAIL}
                    accessibilityLabel={AUTOMATION_TEST_ID.SOCIAL_BUTTON_EMAIL}>
                    <SocialLoginButton
                        title={strings['label.email']}
                        icon={<EmailIcon />}
                        onPress={() => {
                            navigation.navigate(NAVIGATION_TYPE.AUTH_SIGN_IN, {
                                signInType: AUTH_TYPES.EMAIL_SIGN_IN,
                                screenType: EMAIL_AUTH_CONSTANTS.ENTER_EMAIL_SCREEN,
                            });
                        }}
                    />
                </View>
                <View
                    style={socialLoginFooterStyle.socialButtonStyle}
                    testID={AUTOMATION_TEST_ID.SOCIAL_BUTTON_FACEBOOK}
                    accessibilityLabel={AUTOMATION_TEST_ID.SOCIAL_BUTTON_FACEBOOK}>
                    <SocialLoginButton
                        title={strings['signin.social.facebook']}
                        icon={<Image source={require('assets/images/social-login-icons/facebook_logo.png')} />}
                        onPress={onFacebookSignin}
                    />
                </View>
                <View
                    style={socialLoginFooterStyle.socialButtonStyle}
                    testID={AUTOMATION_TEST_ID.SOCIAL_BUTTON_GOOGLE}
                    accessibilityLabel={AUTOMATION_TEST_ID.SOCIAL_BUTTON_GOOGLE}>
                    <SocialLoginButton
                        title={strings['signin.social.google']}
                        icon={<GoogleIcon />}
                        onPress={onGoogleSignin}
                    />
                </View>
            </View>
        </View>
    );
}

export interface SocialLoginFooterProps {
    onSignProgress: () => void;
    onSigninSuccess: (id: string, provider: string) => void;
    onSigninError: (error: string, provider: string) => void;
}
