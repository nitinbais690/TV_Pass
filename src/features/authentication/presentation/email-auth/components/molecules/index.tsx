import { useLocalization } from 'contexts/LocalizationContext';
import PrimaryButton from 'core/presentation/components/atoms/PrimaryButton';
import { TextinputBox } from 'core/presentation/components/atoms/TextinputBox';
import PasswordInputBox from 'core/presentation/components/atoms/PasswordInputBox';
import InputErrorView from 'core/presentation/components/atoms/InputErrorView';
import { appFontStyle } from 'core/styles/AppStyles';
import { EMAIL_AUTH_CONSTANTS } from 'features/authentication/utils/auth-constants';
import React from 'react';
import { Text, View } from 'react-native';
import { useAppPreferencesState } from 'utils/AppPreferencesContext';
import { emailAuthViewStyle } from './styles';
import ForwordIcon from 'assets/images/ic_forword_white.svg';
import { NAVIGATION_TYPE } from 'screens/Navigation/NavigationConstants';
import { AUTOMATION_TEST_ID } from 'features/authentication/presentation/automation-ids';

export default function EmailAuthView(props: EmailAuthViewProps) {
    const { strings } = useLocalization();
    const appPref = useAppPreferencesState();
    const { appTheme } = appPref;
    let { appColors } = appTheme && appTheme(appPref);
    const style = emailAuthViewStyle(appColors);

    return (
        <View>
            <Text style={[appFontStyle.subTitle, style.emailTitleeStyle]}>{strings['header.email_id']}</Text>

            <View style={style.inputBoxStyle}>
                <TextinputBox
                    value={props.email}
                    onChangeText={props.onChangeEmail}
                    autoCapitalize={'none'}
                    keyboardType={'email-address'}
                    onBlur={props.onEmailBlur}
                    textInputBoxTestID={AUTOMATION_TEST_ID.TEXT_INPUT_BOX_EMAIL_ID}
                    textInputBoxAccessibilityLabel={AUTOMATION_TEST_ID.TEXT_INPUT_BOX_EMAIL_ID}
                />
            </View>
            {props.emailError && <InputErrorView errorText={props.emailError} style={style.inputErrorStyle} />}
            {props.screenType !== EMAIL_AUTH_CONSTANTS.ENTER_EMAIL_SCREEN && (
                <View>
                    <Text style={[appFontStyle.subTitle, style.passwordTitleStyle]}>
                        {strings['header.enter_password']}
                    </Text>

                    <View style={style.inputBoxStyle}>
                        <PasswordInputBox
                            maxLength={60}
                            passwordValue={props.password}
                            onTextChange={props.onChangePassword}
                            onBlur={props.onPasswordBlur}
                            textInputBoxTestID={AUTOMATION_TEST_ID.TEXT_INPUT_BOX_PASSWORD}
                            textInputBoxAccessibilityLabel={AUTOMATION_TEST_ID.TEXT_INPUT_BOX_PASSWORD}
                        />
                    </View>
                    {props.passwordError && (
                        <InputErrorView errorText={props.passwordError} style={style.inputErrorStyle} />
                    )}
                </View>
            )}

            {props.screenType === EMAIL_AUTH_CONSTANTS.SIGN_IN_SCREEN && (
                <View
                    style={style.forgotPasswordContainer}
                    testID={AUTOMATION_TEST_ID.FORGOT_PASSWORD}
                    accessibilityLabel={AUTOMATION_TEST_ID.FORGOT_PASSWORD}>
                    <Text
                        style={[appFontStyle.body3, style.forgotPasswordLb]}
                        onPress={() => {
                            props.navigation.navigate(NAVIGATION_TYPE.AUTH_FORGOT_PASSWORD_RESET);
                        }}>
                        {strings['signin.forgot_your_password']}
                    </Text>
                    <ForwordIcon />
                </View>
            )}

            <PrimaryButton
                disabled={props.isLoading}
                containerStyle={style.buttonstyle}
                title={getButonName(props.screenType, strings)}
                onPress={props.onPress}
                primaryTestID={AUTOMATION_TEST_ID.PRIMARY_BUTTON}
                primaryAccessibilityLabel={AUTOMATION_TEST_ID.PRIMARY_BUTTON}
            />
        </View>
    );
}

function getButonName(screenType: string, strings: any) {
    var buttonName;
    if (screenType === EMAIL_AUTH_CONSTANTS.SIGN_IN_SCREEN) {
        buttonName = strings.sign_in;
    } else if (screenType === EMAIL_AUTH_CONSTANTS.SIGN_UP_SCREEN) {
        buttonName = strings['signup.title'];
    } else {
        buttonName = strings.next;
    }
    return buttonName;
}

interface EmailAuthViewProps {
    screenType: string;
    onPress: () => void;
    onEmailBlur?: () => void;
    onPasswordBlur?: () => void;
    isLoading: boolean;
    navigation: any;
    onChangeEmail: (text: string) => void;
    onChangePassword: (text: string) => void;
    email: string;
    password: string;
    emailError: string;
    passwordError: string;
}
