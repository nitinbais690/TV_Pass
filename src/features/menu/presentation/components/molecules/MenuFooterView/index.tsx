import React from 'react';
import { useLocalization } from 'contexts/LocalizationContext';
import PrimaryButton from 'core/presentation/components/atoms/PrimaryButton';
import SecondaryButton from 'core/presentation/components/atoms/SecondaryButton';
import { View, Text } from 'react-native';
import { useAuth } from 'contexts/AuthContextProvider';
import useAppColors from 'core/presentation/hooks/use-app-colors';
import MenuFooterViewStyle from './style';
import DeviceInfo from 'react-native-device-info';
import { NAVIGATION_TYPE } from 'screens/Navigation/NavigationConstants';
import { useNavigation } from '@react-navigation/native';
import { EMAIL_AUTH_CONSTANTS, AUTH_TYPES } from 'features/authentication/utils/auth-constants';
import { AUTOMATION_TEST_ID } from 'features/menu/presentation/automation-ids';

/*
   Renders Menu Footer view - buttons with app version
 */
export function MenuFooterView() {
    const { strings }: any = useLocalization();
    const styles = MenuFooterViewStyle(useAppColors());
    const navigation = useNavigation();

    const readableVersion = DeviceInfo.getReadableVersion();
    const { userType } = useAuth();
    const isLoggedIn = userType === 'LOGGED_IN' || userType === 'SUBSCRIBED';
    return (
        <View style={styles.container}>
            {!isLoggedIn && (
                <SecondaryButton
                    containerStyle={[styles.buttonStyle, styles.marginBottom13]}
                    title={strings.menu_sign_in_btn_label}
                    onPress={() => {
                        navigation.navigate(NAVIGATION_TYPE.AUTH_SIGN_IN, {
                            signInType: AUTH_TYPES.EMAIL_SIGN_IN,
                            screenType: EMAIL_AUTH_CONSTANTS.ENTER_EMAIL_SCREEN,
                        });
                    }}
                    secondaryTestID={AUTOMATION_TEST_ID.SIGN_IN_BUTTON}
                    secondaryAccessibilityLabel={AUTOMATION_TEST_ID.SIGN_IN_BUTTON}
                />
            )}
            <PrimaryButton
                containerStyle={styles.buttonStyle}
                title={strings.menu_subscribe_now_btn_label}
                onPress={() => {}}
                primaryTestID={AUTOMATION_TEST_ID.SUBSCRIBE_NOW_BUTTON}
                primaryAccessibilityLabel={AUTOMATION_TEST_ID.SUBSCRIBE_NOW_BUTTON}
            />

            <Text style={styles.versionTextStyle}>
                {strings['settings.version']} {readableVersion}
            </Text>
        </View>
    );
}
