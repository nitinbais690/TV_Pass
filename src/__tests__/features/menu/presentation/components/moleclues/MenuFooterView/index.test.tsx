import React from 'react';
import { render, toJSON } from '@testing-library/react-native';
import { MenuFooterView } from 'features/menu/presentation/components/molecules/MenuFooterView';

jest.mock('core/presentation/hooks/use-app-colors');
jest.mock('contexts/AuthContextProvider');

describe('Menu Footer view test', () => {
    it('Menu Footer view render correctly', () => {
        const { container } = render(<MenuFooterView />);
        expect(toJSON(container)).toMatchSnapshot();
    });
});
