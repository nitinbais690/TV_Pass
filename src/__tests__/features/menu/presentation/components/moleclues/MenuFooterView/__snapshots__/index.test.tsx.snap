// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Footer view test Menu Footer view render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "left": 0,
        "right": 0,
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 3,
          "paddingHorizontal": 28,
          "width": "100%",
        }
      }
    >
      <View
        accessibilityLabel="SUBSCRIBE_NOW_BUTTON"
        accessible={true}
        focusable={true}
        hasTVPreferredFocus={true}
        isTVSelectable={true}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="SUBSCRIBE_NOW_BUTTON"
      >
        <BVLinearGradient
          colors={
            Array [
              4294929710,
              4290124297,
            ]
          }
          endPoint={
            Object {
              "x": 0.5,
              "y": 1,
            }
          }
          locations={null}
          startPoint={
            Object {
              "x": 0.5,
              "y": 0,
            }
          }
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#3B4046",
              "borderColor": "#2089dc",
              "borderRadius": 35,
              "borderWidth": 0,
              "elevation": 0,
              "flexDirection": "row",
              "justifyContent": "center",
              "padding": 8,
              "paddingHorizontal": 8,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontFamily": "ProximaNova-Regular",
                "fontSize": 18,
                "fontWeight": "600",
                "paddingBottom": 1,
                "paddingTop": 2,
                "textAlign": "center",
              }
            }
          >
            Subscribe now
          </Text>
        </BVLinearGradient>
      </View>
    </View>
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "fontWeight": "600",
          "marginVertical": 20,
          "opacity": 0.3,
        }
      }
    >
      Version
       
    </Text>
  </View>
</View>
`;
